k8s_pod_watcher.go

```diff
@@ -166,6 +166,7 @@ func (p *PodWatcher) initialPods(ctx context.Context) (int, string, error) {
 			cb(ctx, &CreatePod{
 				name: pod.Name,
 				IP:   &net.IPAddr{IP: ipaddr},
+				Def:  &initPods.Items[i],
```

Comment:
Could we use  a pointer to `pod` here instead since we are already ranging over `initPods.Items`?

Reply:
Heh -- I asked the same question :)  Because it's the loop variable you'd end up clobbering the pointer on subsequent loop iterations.  That's why I added those additional tests to make sure that the pod that was populated in `Def` was actually the one that we're expecting...